qcodes/instrument_drivers/Keysight/N52xx.py

```diff
@@ -196,30 +218,34 @@ def write(self, cmd: str) -> None:
         """
         Select correct trace before querying
         """
-        self.root_instrument.active_trace(self.trace)
+        self.root_instrument.active_trace(self.trace_num)
         super().write(cmd)
 
     def ask(self, cmd: str) -> str:
         """
         Select correct trace before querying
         """
-        self.root_instrument.active_trace(self.trace)
+        self.root_instrument.active_trace(self.trace_num)
         return super().ask(cmd)
 
-    @staticmethod
-    def parse_paramstring(paramspec: str) -> Tuple[str, str, str]:
+    def parse_paramstring(self) -> Tuple[str, str]:
         """
         Parse parameter specification from PNA
         """
+        paramspec = self.root_instrument.ask("CALC:PAR:CAT:EXT?")
         paramspec = paramspec.strip('"')
-        ch, param, trnum = re.findall(r"CH(\d+)_(S\d+)_(\d+)", paramspec)[0]
-        return ch, param, trnum
+        specs = paramspec.split(',')
+        for spec_ind in range(len(specs)//2):
+            name, param = specs[spec_ind*2:(spec_ind+1)*2]
+            if name == self.trace_name:
+                return name, param
+        raise RuntimeError("Can't find selected trace on the PNA")
 
     def _Sparam(self, paramspec: str) -> str:
```

Comment:
now this method has an unused argument. Please, fix. Also, is there any use for other elements of `self.parse_paramstring()` (not only `[1]` that is used in this method)? And where are they documented?

Reply:
The parameter is now removed, as is the `[1]` array index. They were `(name, param)` pairs, so there is nothing more, but this is also clearer now.